index.d.ts

```diff
@@ -117,1865 +118,1869 @@ export declare interface Tracer extends opentracing.Tracer {
    * @param {User} user Properties of the authenticated user. Accepts custom fields.
    * @returns {Tracer} The Tracer instance for chaining.
    */
-  setUser (user: User): Tracer;
-
-  appsec: Appsec;
-
-  TracerProvider: opentelemetry.TracerProvider;
-
-  dogstatsd: DogStatsD;
-}
-
-export declare interface TraceOptions extends Analyzable {
-  /**
-   * The resource you are tracing. The resource name must not be longer than
-   * 5000 characters.
-   */
-  resource?: string,
+  setUser (user: tracer.User): Tracer;
 
-  /**
-   * The service you are tracing. The service name must not be longer than
-   * 100 characters.
-   */
-  service?: string,
+  appsec: tracer.Appsec;
 
-  /**
-   * The type of request.
-   */
-  type?: string
-}
+  TracerProvider: tracer.opentelemetry.TracerProvider;
 
-/**
- * Span represents a logical unit of work as part of a broader Trace.
- * Examples of span might include remote procedure calls or a in-process
- * function calls to sub-components. A Trace has a single, top-level "root"
- * Span that in turn may have zero or more child Spans, which in turn may
- * have children.
- */
-export declare interface Span extends opentracing.Span {
-  context (): SpanContext;
+  dogstatsd: tracer.DogStatsD;
 }
 
-/**
- * SpanContext represents Span state that must propagate to descendant Spans
- * and across process boundaries.
- *
- * SpanContext is logically divided into two pieces: the user-level "Baggage"
- * (see setBaggageItem and getBaggageItem) that propagates across Span
- * boundaries and any Tracer-implementation-specific fields that are needed to
- * identify or otherwise contextualize the associated Span instance (e.g., a
- * <trace_id, span_id, sampled> tuple).
- */
-export declare interface SpanContext extends opentracing.SpanContext {
-  /**
-   * Returns the string representation of the internal trace ID.
-   */
-  toTraceId (): string;
-
-  /**
-   * Returns the string representation of the internal span ID.
-   */
-  toSpanId (): string;
+declare namespace tracer {
+  export type SpanOptions = opentracing.SpanOptions;
```

Comment:
This is done this way to force TS to see this as a type import and not a value import (which isn't allowed in a namespace)

Reply:
(as opposed to including it in the below export)